akka-docs/src/main/paradox/typed/actor-discovery.md

```diff
@@ -14,29 +14,37 @@ To use Akka Actor Typed, you must add the following dependency in your project:
+1. that it receives in the message 
+2. that it already had before it received the message 
+3. that it creates while processing the message. “
+
+Similarly, Akka has three general ways to obtain Actor references, but also adds a discovery method
+1. By @ref:[creating actors](actor-lifecycle.md#creating-actors) 
+2. Passed as constructor parameters to an actor on its creation
+3. Passed as parameters of a message received by an actor
+4. By discovery using the @ref:[Receptionist](#receptionist)
 
 ## Receptionist
 
-When an actor needs to be discovered by another actor but you are unable to put a reference to it in an incoming message,
-you can use the `Receptionist`. You register the specific actors that should be discoverable 
-from other nodes in the local `Receptionist` instance. The API of the receptionist is also based on actor messages. 
-This registry of actor references is then automatically distributed to all other nodes in the cluster. 
-You can lookup such actors with the key that was used when they were registered. The reply to such a `Find` request is 
-a `Listing`, which contains a `Set` of actor references that are registered for the key. Note that several actors can be 
+The `Receptionist` is useful when specific actors need to be discoverable by other actors and you need to bootstrap the interaction. 
+For example when they need to be discoverable by actors running either on different nodes in the Cluster, 
+or if the above three general methods for @ref:[obtaining Actor references](#obtaining-actor-references) are not applicable.
+
+An actor can be registered with a `ServiceKey` by sending a `Receptionist.Register` message to the local `ActorSystem` `Receptionist` instance. 
+When using Akka Cluster this registry of actor references is automatically distributed to all other nodes in the cluster. 
+Registered actors can then be looked up by `ServiceKey` using a `Receptionist.Find` request.
+The reply to a `Find` request is a `Receptionist.Listing` containing a `Set` of actor references registered for the key. Note that several actors can be 
 registered to the same key.
 
 The registry is dynamic. New actors can be registered during the lifecycle of the system. Entries are removed when 
-registered actors are stopped, manually deregistered or the node they live on is removed from the @ref:[Cluster](cluster.md). 
-To facilitate this dynamic aspect you can also subscribe to changes with the `Receptionist.Subscribe` message. It will send 
-`Listing` messages to the subscriber when entries for a key are changed.
+registered actors are stopped, manually de-registered or the node they live on is removed from the @ref:[Cluster](cluster.md). 
+Because it is dynamic, changes for a`ServiceKey` can be subscribed to with the `Receptionist.Subscribe` message.
+and a `Receptionist.Listing` message is sent to those subscribers as changes occur.
```

Comment:
We should probably mention `GroupRouter` immediately here, "Keeping track of available services for a key and routing messages to one of them, which is one of the most common use cases is covered by the @ref[GroupRouter] but there are use cases when keeping track of registered services are needed for more complex logic" something something